asserts/asserts.go

```diff
@@ -415,6 +415,38 @@ func (ref *Ref) Resolve(find func(assertType *AssertionType, headers map[string]
 	return find(ref.Type, headers)
 }
 
+// SequenceKey returns the sequence key and the sequence derived from the primary key
+// of the assertion reference, if the assertion type is sequence forming.
+func (ref *Ref) SequenceKey() ([]string, int, error) {
+	if !ref.Type.SequenceForming() {
+		return nil, 0, fmt.Errorf("%q must be a sequence-forming assertion type", ref.Type.Name)
+	}
+
+	if len(ref.PrimaryKey) == len(ref.Type.PrimaryKey)-1 {
+		return ref.PrimaryKey, -1, nil
+	}
```

Comment:
Refs are not really meant to have partial primary keys. OTOH there is AtSequence that might be useful?